apps/angular/src/core/services/user.service.ts

```
@@ -0,0 +1,109 @@
+
+  /** Errors related to email. */
+  readonly email?: string[];
+
+  /** Errors related to password. */
+  readonly password?: string[];
+}
+
+/** Registration errors. */
+export interface RegistrationErrors {
+
+  /** Errors related to email. */
+  readonly email?: string[];
+
+  /** Errors related to first name. */
+  readonly firstName?: string[];
+
+  /** Errors related to last name. */
+  readonly lastName?: string[];
+
+  /** Errors related to password. */
+  readonly password?: string[];
+}
+
+/** Authorization service. */
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+
+  private readonly userUrl: URL;
+
+  public constructor(
+    config: AppConfigService,
+    private readonly http: HttpClient,
+    private readonly authService: AuthService,
+    private readonly tokenService: TokenService,
+  ) {
+    this.userUrl = new URL(`users/profile/`, config.apiUrl);
+  }
+
+  /**
+   * Log шn.
+   * @param loginData Data required for login..
+   */
+  public login(loginData: LoginData): Observable<void | RegistrationErrors> {
+    return this.authService.login(loginData).pipe(
+      catchHttpErrorResponse(error => throwError(() => this.createError(error))),
+    );
+  }
+
+  /**
+   * Registers user.
+   * @param registrationData Data required for registration.
+   */
+  public register(registrationData: RegistrationData): Observable<void | RegistrationErrors> {
+    return this.authService.register(registrationData).pipe(
+      catchHttpErrorResponse(error => throwError(() => this.createError(error))),
+    );
+  }
+
+  /**
+   * Requests to the server to get user profile.
+   * If an error occurs, the user will be null.
+   * The error may be if the user is not authorized.
+   */
+  public fetchUser(): Observable<User | null> {
+    return this.tokenService.get().pipe(
+      switchMap(() => this.http.get<UserDto>(this.userUrl.toString())),
+      map(userDto => UserMapper.fromDto(userDto)),
+      catchError(() => of(null)),
```

Usually, silencing errors is a bad practice that may cost you hours of debugging. So, in such cases, I recommend at least having a warning and specify why you ignore the error